lightning/src/util/events.rs

```diff
@@ -67,6 +68,38 @@ pub enum PaymentPurpose {
 	SpontaneousPayment(PaymentPreimage),
 }
 
+#[derive(Clone, Debug)]
+/// Some information provided on the closure source of the channel halting.
+pub enum ClosureReason {
+	/// Closure generated from receiving a peer error message by ChannelManager::handle_error
+	CounterpartyForceClosed {
+		/// The error is coming from the peer, there *might* be a human-readable msg
+		peer_msg: Option<String>,
+	},
+	/// Closure generated from ChannelManager::force_close_channel
+	HolderForceClosed,
+	/// Closure generated from receiving a peer's ClosingSigned message. Note the shutdown
+	/// sequence might have been initially initiated by us.
+	CooperativeClosure,
+	/// Closure generated from receiving chain::Watch's CommitmentTxBroadcast event.
+	CommitmentTxBroadcasted,
+	/// Closure generated from processing an event, likely a HTLC forward/relay/reception.
+	ProcessingError {
+		err: String,
+	},
+	/// Closure generated from ChannelManager::peer_disconnected.
```

Comment:
Should mention the `no_connection_possible` flag and describe why this method is called, eg "The `PeerManager` informed us that we've disconnected from the peer and that it is unlikely we'll be able to connect to the peer, most likely because we have incompatible features and the peer, or we, require features which we, or the peer, do not support."